cmd/charmd/main.go

```diff
@@ -105,6 +116,17 @@ func serve(confPath string) error {
 	return http.ListenAndServe(conf.APIAddr, debug.Handler("", server))
 }
 
+func addPublicKey(ring *bakery.PublicKeyRing, loc string, key *bakery.PublicKey) error {
+	if key != nil {
+		return ring.AddPublicKeyForLocation(loc, false, key)
+	}
+	pubKey, err := httpbakery.PublicKeyForLocation(http.DefaultClient, loc)
```

Comment:
@alesstimec I think @rogpeppe is referring to comments he made on #491 after it landed yesterday.